synapse/util/iterutils.py

```diff
@@ -23,19 +23,18 @@
     Mapping,
     Sequence,
     Set,
-    Tuple,
     TypeVar,
 )
 
 T = TypeVar("T")
 
 
-def batch_iter(iterable: Iterable[T], size: int) -> Iterator[Tuple[T]]:
+def batch_iter(iterable: Iterable[T], size: int) -> Iterator[Collection[T]]:
```

Comment:
this is fine, but I'm a bit curious about why it is needed. it definitely returns an iterator over `tuple`s.

Reply:
`Tuple[T]` means a tuple of a single element of type `T`, but the returned tuples return are (likely) longer than a single element.

Reply:
FWIW `Tuple[T, ...]` means a variable length tuple of type `T`

Reply:
I thought I tried that and it didn't work...I can try again though.

Reply:
hrrrm. I think we may have stuff that relies on it being specifically tuples, so I'm a bit hesitant to have an annotation that says the opposite, even if mypy is happy with it. As erik says, I'd expect `Iterator[Tuple[T...]]` to work.

Reply:
Thanks for the feedback! It seems `Iteraor[Tuple[T, ...]]` worked!